### ISSUE TITLE
We have a lot of transactions stuck on the chain with below < 1 tx fee. 

### ISSUE THREAD

**Issue by ara1313** [dogecoin/dogecoin#1652]
@shibe2 @langerhans We are in the situation where we have txs that are unconfirmed on the chain with the fee below 1.

It seems that we can't use bumpfee for the transactions that already went out and sitting unconfirmed on chain. How would you recommend to get those tx confirmed?

example: https://dogechain.info/tx/38be6b764da587333892bdbb0f20561f9ba2880f50d50ec3ce77bbae0660f8be

We also have large number of transactions stuck in the mempool with the fee below 1. What should we do with the txs stuck in the mempool?

After we upgrade to the 1.14.2, what happens to the transactions in the mempool? Is it going to reload?

**Comment by shibe2** [comment]
@ara1313 As I said, I believe, such transaction should be eventually confirmed. You can try things like CPFP, re-sending coins with appropriate fee, sending transactions directly to miners.

Maybe come to Dogecoin [Discord](https://invite.gg/dogecoin) or [reddit](http://www.reddit.com/r/dogecoin) where more active Dogecoiners may be available to help you.

**Comment by patricklodder** [comment]
- Your tx will not be confirmed unless it gets relayed to a miner node AND they include it.
- CPFP does not work in this case, because the child tx only gets accepted by nodes that have the parent (eg nodes that were directly connected to yours) but if the parent did not propagate, the child ends up being a spend of an invalid utxo and thus will not be relayed far. So currently if the transaction does not get directly relayed to a miner node by the originating node, this solution does not work.
- Mempool expiry time is currently defaulting to 2 weeks, so most nodes will keep your tx in their mempool for that long. I realize that that's very non-optimal 😕

Re-sending is more viable, especially since you seem to re-use your change address. What you could do (with Dogecoin core, examples are for a linux dogecoind):
1. Dump all your txs into a backup, eg: 
```
dogecoin-cli getrawmempool |jq -r '.[]' |xargs -n 1 dogecoin-cli getrawtransaction >> my_mempool.txt
```
2. Make note of all the peers your node is currently connected to with `dogecoin-cli getpeerinfo`, those will have your tx in mempool and will not accept replacement tx.
3. Shut down your node.
4. Delete your local mempool, eg: `rm ~/.dogecoin/mempool.dat`
5. If you run with wallet enabled and those tx were made by the wallet and not by `sendrawtransaction`, restart your node with zapwallet, eg: `dogecoind -zapwallettx`, otherwise just restart the node.
6. Add some additional nodes (you can manually discover some with `dig seed.multidoge.org`)
7. For each of the backed-up tx that are yours, adjust the fee and resend. Manually, that would be:
    - count the bytes
    - adjust the fee, eg: `dogecoin-tx <tx_hex> delout=<change output index number> outaddr=<new change>:<change addr>`
    - re-sign it with `dogecoin-cli signrawtransaction <tx_hex> null '["<privatekey_wif>"]'`
    - check if the new byte count does not exceed the fee paid
    - resend it with `dogecoin-cli sendrawtransaction <signed_tx_hex>`

If you have a lot of re-spent change outputs (the example you have shown doesn't respend, but do consider this) it may be quicker to create a new (set of) tx instead of step 7. If you're unsure about the fee needed, RBF opt-in works (by setting a sequence number `< 0xffffffff - 1` on your tx inputs)

**Comment by shibe2** [comment]
@patricklodder If CPFP is supported at all, it is possible to send both the stuck and child transaction directly to miners.

**Comment by patricklodder** [comment]
> If CPFP is supported at all, it is possible to send both the stuck and child transaction directly to miners. 

If you have a way to do that, yes. If you're chaining tx (new change address per tx) it may be easier that way, if you're not, it's probably more work and creating a new tx with sufficient fee is then probably the easiest way forward.

**Comment by richiela** [comment]
@patricklodder Thanks for the detailed response.  Three questions.

It sounds like there is a way to push these through if someone could point us to a miner that we could pay to include them?  Is this true?  Do they just have to use an older version of the daemon so it can accept v3 blocks?

If we did use sendrawtransaction, what's the easiest way to just delete all the 0 confirmed transactions? In the past in situations like this, we've had to dump all the keys, reimport, rescan to get to a clean state.

Once we remove all the tx's from the wallet, we can just requeue and resend them all with the correct fee; the concern is during the 2 week its in the mempool, some of them somehow get mined and included at which point, we have double paid someone.  I really want to avoid having to figure out all the inputs and creating a transaction just to doublespend them to avoid this situation.

**Comment by shibe2** [comment]
@richiela

Block version is unrelated to fee policy. And also, mining with older version will not get transactions confirmed, the block will be rejected by other miners, I think.

If you used sendrawtransaction and it's not in the node's wallet, you just have to update to latest version of Dogecoin Core and delete mempool.dat.

**Comment by richiela** [comment]
So what’s the best way to get the current transactions mined?  I’m happy to pay for the hash but need to understand what to ask for.

**Comment by shibe2** [comment]
Miner can decrease `minrelaytxfee`. Hopefully it will work.

**Comment by patricklodder** [comment]
> It sounds like there is a way to push these through if someone could point us to a miner that we could pay to include them? Is this true? 

This is possible and true, yes.

> Do they just have to use an older version of the daemon so it can accept v3 blocks?

No, that's not needed. At the most you'd need to create a "payment transaction" on the top that will re-spend all your change outputs from the currently stuck transactions and pay the agreed fee.
 
> If we did use sendrawtransaction, what's the easiest way to just delete all the 0 confirmed transactions? In the past in situations like this, we've had to dump all the keys, reimport, rescan to get to a clean state.

Step 1,3,4 and 5 that I outlined above are sufficient for that. You don't need to dump and reimport keys, `-zapwallettx`, has the desired outcome (that your unconfirmed wallet transactions are removed.)

>Once we remove all the tx's from the wallet, we can just requeue and resend them all with the correct fee; the concern is during the 2 week its in the mempool, some of them somehow get mined and included at which point, we have double paid someone. I really want to avoid having to figure out all the inputs and creating a transaction just to doublespend them to avoid this situation.

You'd have to re-spend the inputs to be sure. I see how this can cause work, perhaps we can make a small script that takes a txid list in, and re-spends them, to automate this. May be of use for others as well.

**Comment by patricklodder** [comment]
> Miner can decrease `minrelaytxfee`. Hopefully it will work.

If the miner just submits the tx to the node(s) that create the block templates, this should be enough.

**Comment by richiela** [comment]
Anyone have any contacts with mining pools?  i've been trying to each out to no avail.

**Comment by langerhans** [comment]
> Anyone have any contacts with mining pools? i've been trying to reach out to a few but to no no avail.

We're trying to get in contact with a few

**Comment by ReverseControl** [comment]
@richiela If you have not resolved your issue, the upcoming release ( that should be out by the end of the week at the latest) will likely help you resolve your issue. 

> Miner can decrease `minrelaytxfee`. Hopefully it will work.

``minrelaytxfee`` will be set to 1/1000 of its current value, so .001 DOGE, furthermore there was a bug with ``minrelaytxfee`` that has been fixed in 1.14.4, so you may set this fee higher and it will do the right thing in all scenarios.

@patricklodder @rnicoll @michilumin We can close this one.
